File: Lab05/python/findzero/findzero.py
import math

def cos_2(X):
    return math.cos(X/2)

def findzero(f, a, b, eps):
    srodek = (a+b)/2

    while (abs(a-b) > eps):
        if (f(a) * f(srodek) < 0.0):
            b = srodek
            srodek = (a+b)/2
        else:
            a = srodek
            srodek = (a+b)/2

    return srodek

File: semestr1/Lab05/c/findzero.c
#include <math.h>
#include "findzero.h"

double cos_2(double X){
    
    return cos(X/2);
}

double findzero(functype f, double a, double b, double eps){
    double srodek = (a+b)/2;

    while (fabs(a-b) > eps){
        if (f(a) * f(srodek) < 0.0){
            b = srodek;
        }
        else{
            a = srodek;
        }
        srodek = (a+b)/2;

    }
    return srodek;
}